### PR TITLE
bundle spid-php-lib/vendor and thereby fix #28

### LINKED ISSUE #28: a bundled copy of spid-php-lib/vendor is missing
spid-php-lib [depends](https://github.com/italia/spid-php-lib/blob/master/composer.json#L16) on https://packagist.org/packages/robrichards/xmlseclibs

currently `spid-php-lib/vendor` is missing therefore the library cannot work

### PR DESCRIPTION
**Pull request by simevo**
- supplies a script to automate this procedure in the future
- documents the bundling procedure in README-DEV.md
- exludes the script and the new readme from the distributables

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
         "email": "paolo.greppi@libpf.com"
     }],
     "require": {
+        "italia/spid-php-lib": "^0.20.0",
```
Comment (michaeltieso) [inline composer.json]: This is not needed because it's included in the main directory.

File: includes/class-spid-core.php [hunk 2]
```diff
@@ -14,6 +14,8 @@
 	die;
 }
 
+require_once dirname(__FILE__) . '/../spid-php-lib/vendor/autoload.php';
```
Comment (michaeltieso) [inline includes/class-spid-core.php]: All of these are being loaded already

File: composer.json (hunk 1 above)
Comment (giuliogatto) [reply composer.json]: I am not sure it's a good idea to remove this: 'hard coding' a class to the repo whilst is being actively updated is dangerous and difficult to maintain (and scale). I like the idea of offering a self contained plugin but there are risks involved. We should find a way to offer both options, maybe automatically selecting from within the plugin the most recent class version.

File: includes/class-spid-core.php (hunk 2 above)
Comment (simevo) [reply includes/class-spid-core.php]: I don't think so. To reproduce:
- merge PR #43 or cherry-pick the commit 7b321bbb1ad61e105324d82170df3eec076d8310
- make sure the certificates are available in `{wordpress root}/spid-conf`
- manually install dependencies in spid-php-lib running `composer install` in there
- load the metadata endpoint (for example: http://localhost:8099/wp-login.php?sso=spid&metadata), get:
```
  | <br />
-- | --
  | <b>Fatal error</b>:  Uncaught Error: Class 'RobRichards\XMLSecLibs\XMLSecurityKey' not found in /var/www/html/wp-content/plugins/spid-wordpress/spid-php-lib/src/Spid/Saml/SignatureUtils.php:19
  | Stack trace:
  | #0 /var/www/html/wp-content/plugins/spid-wordpress/spid-php-lib/src/Spid/Saml.php(127): Italia\Spid\Spid\Saml\SignatureUtils::signXml('&lt;md:EntityDescr...', Array)
  | #1 [internal function]: Italia\Spid\Spid\Saml-&gt;getSPMetadata()
  | #2 /var/www/html/wp-content/plugins/spid-wordpress/spid-php-lib/src/Sp.php(36): call_user_func_array(Array, Array)
  | #3 /var/www/html/wp-content/plugins/spid-wordpress/includes/class-spid-core.php(98): Italia\Spid\Sp-&gt;__call('getSPMetadata', Array)
  | #4 /var/www/html/wp-includes/class-wp-hook.php(286): SPID_Core-&gt;filterAuthenticate(Object(WP_Error), '', '')
  | #5 /var/www/html/wp-includes/plugin.php(203): WP_Hook-&gt;apply_filters(Object(WP_Error), Array)
  | #6 /var/www/html/wp-includes/pluggable.php(532): apply_filters('authenticate', NULL, '', '')
  | #7 /var/www/html/wp-includes/user.php(90): wp_authenticate('', '')
  | #8  in <b>/var/www/html/wp-content/plugins/spid-wordpress/spid-php-lib/src/Spid/Saml/SignatureUtils.php</b> on line <b>19</b><br />
```

File: composer.json (hunk 1 above)
Comment (michaeltieso) [reply composer.json]: I'm sorry. I'm having trouble understanding. Could you clarify? Any future updates of the plugin need to be updated manually and pushed into WP.org. Whether that's through Composer or by move/replace, it doesn't matter. This only applies to the developer working on this plugin and does not do anything for the end user using this plugin.

Comment (giuliogatto) [reply composer.json]: So there won't be any other channel of distribution for the plugin? Only a manual update into WP.org? IE: not download a .zip from github?

Comment (michaeltieso) [reply composer.json]: There might be but doesn't make sense not to distribute it on WP.org. This makes it an even stronger case why we should not require Composer to use this plugin.

**Pull request merged**